### ISSUE TITLE
`haskell_doc` splits generated documentation across multiple directories with mangled names

### ISSUE THREAD

**Issue by gdeest** [tweag/rules_haskell#232]
We want to generate Haddock documentation on CI, and upload the output as part of CI artifacts. `rules_haskell` does not currently support this use case very well:

- Documentation is split across multiple directories in `./bazel-bin` (one per target). In the example posted in #231 , for example, documentation will be found in `./bazel-bin/libA/something` and `./bazel-bin/libB/other-thing`. This is inconvenient, as it means that I need to manually retrieve all these directories to build the upload archive. More importantly, I may very well miss a dependence (or forget to add it later on when it is introduced) and end up with only partial documentation. 

- The name of the directories where documentation can be found are a bit cryptic, undocumented, and it looks like they may change in the future without warning, making any attempt at automation fragile. Currently, using once again the example in #231 , those will be: `./bazel-bin/libB/doc-libBZSlibB-1.0.0/` and `./bazel-bin/libB/doc-libBZSlibB-1.0.0/` (where does the `1.0.0` ? It seems to be hard-coded by `rules_haskell`).

To put it simply, I'm looking for a way to generate self-contained documentation from a single root target, including all dependencies.
